apollo-router/src/plugins/authentication/subgraph.rs

```diff
@@ -192,12 +198,100 @@ pub(crate) struct SigningParams {
 
 #[derive(Clone)]
 pub(crate) struct SigningParamsConfig {
-    credentials_provider: Arc<dyn ProvideCredentials>,
+    credentials_provider: CredentialsProvider,
     region: Region,
     service_name: String,
     subgraph_name: String,
 }
 
+#[derive(Clone, Debug)]
+struct CredentialsProvider {
+    credentials: Arc<RwLock<Credentials>>,
```

Comment:
How should we handle failures to refresh credentials for X period of time, exceeding the validity of those credentials?

Reply:
Good question, `refresh_credentials()` can be called manually if we end up in such a situation, the plan is to intercept sts 401s and call refresh_credentials(), so this + the refresh mechanism should give us enough guardrails.